### ISSUE TITLE
[Design] Profile: Create a generic error state for when direct deposit is down

### ISSUE THREAD

**Issue by Samara-Strauss** [department-of-veterans-affairs/va.gov-team#5959]
## User Story or Problem Statement

As a user, I need to know if VA.gov is unable to retrieve my direct deposit information so I know what is going on if I don't see it in my profile.

## Additional background

On 2/18/20, we discovered that there is no generic error state in the profile if we are unable to retrieve direct deposit info because of an API or other issue. Here is a PR for this issue that puts a bandaid on things on the code end: https://github.com/department-of-veterans-affairs/vets-website/pull/11716

From Erik:

we need to handle the situation where the GET ppiu/payment_information endpoint returns an error:

```
{
  "errors": [
    {
      "title": "Bad Gateway",
      "detail": "Received an an invalid response from the upstream server",
      "code": "EVSS502",
      "source": "EVSS::PPIU::Service",
      "status": "502"
    }
  ]
}
```

In other words, we need to show an error if we can't retrieve direct deposit information so users know why they aren't seeing this feature. This message needs to be generic (eg. "we may not be able to load all sections of the profile") because we won't know who should see DD or not if the API is down or there is something wrong with our connection.

## Tasks

- [x] Create an error message for the profile if we can't retrieve direct deposit information.
- [x] Work with copy to get the message approved

## Acceptance Criteria

- [x] Error message has been created and copy approved.

**Comment by tressaellen** [comment]
@erikphansen do we need to mention direct deposit at all in the messaging? Or can it be pretty vague?  Also, what is the timeline for this kind of error?

![Screen Shot 2020-06-26 at 2.48.07 PM.png](https://images.zenhubusercontent.com/5dfa65bbbd078d1699e52294/a6bdea66-c73a-4c11-8c9c-dadd3cb4550a)

**Comment by erikphansen** [comment]
The current messaging is shown below:
![image.png](https://images.zenhubusercontent.com/5b4d0309a8d8be5bda39303e/5bc77bfe-891d-45ba-a72a-13189025de17)
The difference now is that we are showing this message at the top of the Personal Info section. I think it might be good to leave this vague, though as it's possible we'll show this if we can't access other parts of the profile as well?

**Comment by Samara-Strauss** [comment]
Is this messaging only for when direct deposit is down, or for when there might be multiple sections down at once? If it's specific and only for the case where direct deposit is down, I would err on the side of more specific with the messaging instead of less specific. As a general UX practice, we should always be more specific if (1) we're able to and (2) if it helps the user understand what's going on in a meaningful way.

TL;DR — If this is *only* for when direct deposit is down, I would keep the "We can't access your payment information right now" message, and maybe consider changing "payment" to "direct deposit" pending Peggy's approval.

**Comment by erikphansen** [comment]
This is only for DD

**Comment by tressaellen** [comment]
<img width="1057" alt="Screen Shot 2020-06-29 at 8 35 06 AM" src="https://user-images.githubusercontent.com/58752772/86019148-b807a180-b9d2-11ea-9f5a-1a9a23b0afb8.png">

If this is only for direct deposit, then I think this should work with updating "payment" to "direct deposit. 

If we are using it for other sections as well, could we do something like the editing modal and insert the section title? The only one where the wording might be weird is with account security...  Are there any use cases where multiple sections might be down and we need to list them in an alert?

**Comment by tressaellen** [comment]
Great - then I'll loop Peggy in. 

@peggygannon - we are working on messaging for a generic error state when direct deposit can't be accessed.  The current messaging is this screenshot:

<img width="712" alt="Screen Shot 2020-06-29 at 9 28 03 AM" src="https://user-images.githubusercontent.com/58752772/86024939-1b490200-b9da-11ea-8fed-bb31b6f992c8.png">


We'd like to change "payment" to "direct deposit", any thoughts on that? Or any other feedback for this messaging?

**We can't access your direct deposit information right now.**

We're sorry.  Something went wrong on our end.  Please refresh this page or try again later.

<img width="1057" alt="Screen Shot 2020-06-29 at 8 35 06 AM" src="https://user-images.githubusercontent.com/58752772/86024675-c3120000-b9d9-11ea-8b77-9300ec90de6e.png">

**Comment by Samara-Strauss** [comment]
> Are there any use cases where multiple sections might be down and we need to list them in an alert?

Sure, in theory, this could happen if multiple systems happen to be down at the same time. 

However, this begs the question: Why are we showing the direct deposit error on the first page of the profile? I'm curious as to the rationale here and why we aren't just showing this on the direct deposit section itself. Not saying that we *shouldn't* take this approach, but genuinely curious as to why this decision was made. That's relevant because if we're showing this error on the main page, what are we doing with other errors? Should we show them on the main page or in their own sections? If we show them in their own sections, there's no need for a catch-all error like [in this comment](https://github.com/department-of-veterans-affairs/va.gov-team/issues/5959#issuecomment-651170428).

**Comment by tressaellen** [comment]
The way I understand it is that we won't know who should or shouldn't be able to see DD, which is why this particular alert would live on the first page. 

> because we won't know who should see DD or not if the API is down or there is something wrong with our connection.

BUT, I agree - I think we definitely need to think about where the other errors should live.  I'm not sure which way would make more sense, I'm leaning towards just putting them in their own particular sections.

**Comment by peggygannon** [comment]
Hi, @tressaellen my only feedback is to remove the period in the header of the message. So the header would read: 

**We can't access your direct deposit information right now**

The rest of the message looks good.  I like using the more specific direct deposit and putting the error on the individual sections.

**Comment by tressaellen** [comment]
Thanks @peggygannon !  I think the issue on this ticket has been resolved - I made a [new ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/10722) to look into the other error states and determine the best course of action.

For this ticket the copy should be:

**We can't access your direct deposit information right now**

We're sorry. Something went wrong on our end. Please refresh this page or try again later.

![Screen Shot 2020-06-29 at 11.35.07 AM.png](https://images.zenhubusercontent.com/5dfa65bbbd078d1699e52294/f7c97f51-d30b-464a-afe6-86ced7f5e123)

**Comment by erikphansen** [comment]
I'll work this final copy into my open PR. Aside from the placement of the alert this has feature parity with Profile v1.

This particular error is tricky. Personally I'm not crazy about specifically mentioning that we can't access direct deposit info right now because, to my knowledge, most VA.gov users shouldn't see that section at all. So it feels weird to be telling people that we can't access a feature they should know nothing about. But the only way to know if someone should have access to the Direct Deposit feature is based on what this failing endpoint tells us!

**Comment by Samara-Strauss** [comment]
> The way I understand it is that we won't know who should or shouldn't be able to see DD, which is why this particular alert would live on the first page.

Got it. This totally makes sense (and I totally forgot that I already knew this).

You might want to think through some different use cases to make sure everything makes sense design-wise if multiple systems are down at once. I don't think we need a catch-all error because for most cases besides direct deposit, we can probably just show the error in the section itself since they aren't conditional in the same way DD is.

**Comment by erikphansen** [comment]
We could pretty easily move this particular alert to the top of every single profile section (in which case it would live above the "Personal and contact info" headline). But I think this is probably the correct place for it. In a scenario where a DD user logs in and goes directly to `profile/direct-deposit` and the Direct Deposit endpoint is down, they'll get redirected to the personal info section and see this error. So, 👍
